crates/wast/src/names.rs

```diff
@@ -51,11 +51,13 @@ impl<'a> Namespace<'a> {
     }
 
     pub fn register_specific(&mut self, name: Id<'a>, index: u32, desc: &str) -> Result<(), Error> {
-        if let Some(_prev) = self.names.insert(name, index) {
-            return Err(Error::new(
-                name.span(),
-                format!("duplicate identifier for {}", desc),
-            ));
+        if let Some(prev) = self.names.insert(name, index) {
+            if prev != index {
```

Comment:
Is this condition still necessary?

Reply:
Yep, imagine if someone writes 
```
(type (struct (field $vt (mut i32)) (field $vt (mut i64))))
```

Adding a test case.

Reply:
I don't believe that motivates this condition because previously that was considered an error as well?

Reply:
You're right. The added test works too.

Reply:
Removing.